File: src/main/java/com/randomperson22/piggymodtest/entities/reindessa/ReindessaModel.java
package com.randomperson22.piggymodtest.entities.reindessa;

import net.minecraft.util.ResourceLocation;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class ReindessaModel extends AnimatedGeoModel<EntityReindessa> {
	
    public ResourceLocation getModelLocation(EntityReindessa entity) {
        return new ResourceLocation("piggymodtest", "geo/reindessa.geo.json");
    }

    public ResourceLocation getTextureLocation(EntityReindessa entity) {
        return new ResourceLocation("piggymodtest", "textures/entities/reindessa.png");
    }

    public ResourceLocation getAnimationFileLocation(EntityReindessa entity) {
        return new ResourceLocation("piggymodtest", "animations/reindessa.animation.json");
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/entities/reindessa/ReindessaModel.kt
package com.randomperson22.piggymodtest.entities.reindessa

import net.minecraft.util.ResourceLocation
import software.bernie.geckolib3.model.AnimatedGeoModel

class ReindessaModel : AnimatedGeoModel<EntityReindessa?>() {
    override fun getModelLocation(entity: EntityReindessa?): ResourceLocation {
        return ResourceLocation("piggymodtest", "geo/reindessa.geo.json")
    }

    override fun getTextureLocation(entity: EntityReindessa?): ResourceLocation {
        return ResourceLocation("piggymodtest", "textures/entities/reindessa.png")
    }

    override fun getAnimationFileLocation(entity: EntityReindessa?): ResourceLocation {
        return ResourceLocation("piggymodtest", "animations/reindessa.animation.json")
    }
}
